### PR TITLE
State and Design Overhaul of Frontend

### PR DESCRIPTION
**Pull request by jackdavidweber**
I was trying to make some small design changes to the frontend but realized that the way we were handling state and styling was not following reactjs best practices. For example:
- we were mixing css stylesheets in when react has its own system for styling components
- we were using classes instead of hooks for managing state despite the [advantages of hooks](https://reactjs.org/docs/hooks-faq.html#should-i-use-hooks-classes-or-a-mix-of-both) 

Additionally, since I wanted to try to [emulate google translate](https://tree.taiga.io/project/jackdavidweber-cjs-capstone/us/94?no-milestone=1) with the frontend design, this seemed like a good opportunity to address both issues at the same time.

### Design
My goal was to emulate the google translate box shown here:
Screen Shot 2020-07-08 at 5 56 46 PM

This is a before (left) and after (right) of the frontend
Screen Shot 2020-07-08 at 5 44 38 PM

### Next Steps

- I did not have time to incorporate the google search feature from [this pull request](https://github.com/cjoshea9/cjs_capstone_frontend/pull/4)
- Need to add in logic that translates when the input / output language is changed (this was not in the original frontend either)

### CODE REVIEW

**Comment by jackdavidweber** [comment]
@swalsh15 and @cjoshea9 would be great if yall could take a look before we have the hosts review since this PR involves significant code changes.

File: src/App.js [hunk 1]
```diff
@@ -68,122 +46,144 @@ class App extends React.Component {
...
-        </FormControl>
-        </div>
+        const requestOptions = {
+        method: 'POST',
+        headers: new Headers({
+            'Content-Type': 'application/x-www-form-urlencoded',
+        }),
+        body: params.toString()
+        };
+        console.log("input", input)
```
Comment (swalsh15) [inline src/App.js:86]: I'd remove the comments before deploying

**Review by swalsh15 — APPROVED** [review]
Looks really nice!

File: src/App.js (hunk 1 above)
Comment (swalsh15) [inline src/App.js]: I'd remove comments before deploying

**Review by cjoshea9 — APPROVED** [review]
This is awesome! It looks really good

File: src/App.js (hunk 1 above)
Comment (jackdavidweber) [reply src/App.js]: will do- thank you!

File: src/App.js [hunk 2]
```diff
@@ -1,62 +1,40 @@
...
-      .then((data) => {
-        this.setState({
-          output: data['response']
-        })
-      })
-  }
-
-  allowTabs() {
+function allowTabs() {
     const textbox = document.getElementById("input-text-area")
```
Comment (g-harel) [inline src/App.js:43]: This should probably use [refs](https://reactjs.org/docs/hooks-reference.html#useref) to be safer.

File: src/App.js [hunk 3]
```diff
@@ -68,122 +46,142 @@ class App extends React.Component {
...
+    const classes = useStyles();
+    let timer = null;
 
-  updateOutputLanguage = (input) => {
-    this.setState({
-        outputLangauge: input.target.value
-    });
-  }
+    // set state
+    const [input, setInput] = useState("")
```
Comment (g-harel) [inline src/App.js]: 🎉 hooks!

File: src/App.js [hunk 4]
```diff
@@ -68,122 +46,142 @@ class App extends React.Component {
...
-          defaultValue={"py"}
-          onChange={this.updateOutputLanguage}
-        >
-          <MenuItem value={"js"}>JavaScript</MenuItem>
-          <MenuItem value={"py"}>Python</MenuItem>
-        </Select>
-        </FormControl>
-        </div>
+        const requestOptions = {
+        method: 'POST',
```
Comment (g-harel) [inline src/App.js]: Nit: indentation.

File: src/App.js [hunk 5]
```diff
@@ -68,122 +46,142 @@ class App extends React.Component {
...
+
+    const handleOutputLanguageChange = (event, value) => {
+        setOutputLanguage(value)
+    } 
+
+    function autoTranslate(value){
+        setInput(value);
+
+        clearTimeout(timer)
+        timer = setTimeout(() => {
```
Comment (g-harel) [inline src/App.js]: Not sure that the timer will be preserved now between renders, I'd consider it part of the component state.

File: src/App.js [hunk 6]
```diff
@@ -68,122 +46,142 @@ class App extends React.Component {
...
+            postRequest()
+        }, 500)   
+    }
+
+    const handleInputChange = (event) => {
+        allowTabs();
```
Comment (g-harel) [inline src/App.js]: Feels odd to me to `allowTabs` in here, IMO would make more sense in the `useEffect` block.

**Review by g-harel — COMMENTED** [review]
Looking great! One small note on design, I initially thought the tabs were all part of the same group, would be nice to add a more visible indication that both halves are distinct.

File: src/App.js (hunk 2 above)
Comment (jackdavidweber) [reply src/App.js:43]: added TODO

**Pull request merged**